src/wasm-ir-builder.h

```diff
@@ -395,16 +398,17 @@ class IRBuilder : public UnifiedExpressionVisitor<IRBuilder, Result<>> {
     static ScopeCtx makeLoop(Loop* loop, Type inputType) {
       return ScopeCtx(LoopScope{loop}, inputType);
     }
-    static ScopeCtx makeTry(Try* tryy, Name originalLabel, Type inputType) {
-      return ScopeCtx(TryScope{tryy, originalLabel}, inputType);
+    static ScopeCtx
+    makeTry(Try* tryy, Name originalLabel, Type inputType, Index index) {
+      return ScopeCtx(TryScope{tryy, originalLabel, index}, inputType);
     }
-    static ScopeCtx makeCatch(ScopeCtx&& scope, Try* tryy) {
-      scope.scope = CatchScope{tryy, scope.getOriginalLabel()};
+    static ScopeCtx makeCatch(ScopeCtx&& scope, Try* tryy, Index index) {
+      scope.scope = CatchScope{tryy, scope.getOriginalLabel(), index};
```

Comment:
Let's remove the `index` parameter here and do the increment automatically, since it never makes sense to create a `Catch` (or `CatchAll`) scope with any index other than one more than the previous index.